lib/luacrypto_acompress.c

```diff
@@ -0,0 +1,350 @@
+		obj->outbuf_len = out_len;
+	}
+
+	sg_init_one(&obj->sg_in, in_buf, in_len);
+	sg_init_one(&obj->sg_out, obj->outbuf, out_len);
+
+	acomp_request_set_params(obj->req, &obj->sg_in, &obj->sg_out, in_len, out_len);
+	acomp_request_set_callback(obj->req, 0, luacrypto_acomp_req_docall, obj);
+
+	/* Table to hold all references */
+	lua_newtable(L);
+	
+	lua_pushvalue(L, 4);
+	lua_setfield(L, -2, "cb");
+	
+	lua_pushvalue(L, 2);
+	lua_setfield(L, -2, "buf");
+	
+	lua_pushvalue(L, 1);
+	lua_setfield(L, -2, "self");
+	
+	obj->refs = luaL_ref(L, LUA_REGISTRYINDEX);
+	
+	obj->busy = true;
+	return 0;
+}
+
+#define LUACRYPTO_ACOMP_REQ_OPERATION(name)							\
+static int luacrypto_acomp_req_##name(lua_State *L)						\
+{												\
+	luacrypto_acomp_req_t *obj = luacrypto_acomp_req_check(L, 1);				\
+	if (obj->busy)										\
+		return luaL_error(L, "request object is busy");					\
+												\
+	size_t in_len;										\
+	const char *in_buf = luaL_checklstring(L, 2, &in_len);					\
+	lunatik_checkbounds(L, 2, in_len, 1, UINT_MAX);						\
+	unsigned int out_len = lunatik_checkuint(L, 3);					\
+	lunatik_checkbounds(L, 3, out_len, 1, UINT_MAX);					\
+	luaL_checktype(L, 4, LUA_TFUNCTION);							\
+												\
+	lunatik_try(L, luacrypto_acomp_req_prepare, obj, L, in_buf, in_len, out_len);	\
+	int ret = crypto_acomp_##name(obj->req);						\
+	if (ret != -EINPROGRESS)								\
+		luacrypto_acomp_req_docall(obj, ret);						\
+	return 0;										\
+}
+
+/***
+* Request object methods.
+* These methods are available on request objects created by `ACOMPRESS:request()`.
+* @type crypto_acomp_req
+*/
+
+/***
+* Compresses data asynchronously using the ACOMPRESS transform.
+* The callback is invoked when the operation completes (either synchronously or asynchronously).
+* @function compress
+* @tparam string data The data to compress.
+* @tparam integer output_size The maximum size of the output buffer.
+* @tparam function callback The callback function to invoke upon completion. It receives two arguments: `err` (integer error code, 0 on success) and `data` (string containing compressed data, or nil on error).
+* @raise Error if the request object is busy, or if parameters are invalid.
+*/
+LUACRYPTO_ACOMP_REQ_OPERATION(compress)
+
+/***
+* Decompresses data asynchronously using the ACOMPRESS transform.
+* The callback is invoked when the operation completes (either synchronously or asynchronously).
+* @function decompress
+* @tparam string data The compressed data to decompress.
+* @tparam integer output_size The maximum size of the output buffer.
+* @tparam function callback The callback function to invoke upon completion. It receives two arguments: `err` (integer error code, 0 on success) and `data` (string containing decompressed data, or nil on error).
+* @raise Error if the request object is busy, or if parameters are invalid.
+*/
+LUACRYPTO_ACOMP_REQ_OPERATION(decompress)
+
+static const luaL_Reg luacrypto_acomp_req_mt[] = {
```

Comment:
let's keep the pattern of prefixing with the full module name.. `luacrypto_acompress_*`